src/coreclr/src/vm/debuginfostore.cpp

```diff
@@ -712,6 +712,22 @@ void CompressDebugInfo::EnumMemoryRegions(CLRDataEnumMemoryFlags flags, PTR_BYTE
     }
     CONTRACTL_END;
```

Comment:
Does this method need a flag that says whether it has the patchpointinfo byte?

Reply:
I don't think so (Andy?) because EEJitManager::GetBoundariesAndVars passes hasFlagByte always true (if  FEATURE_ON_STACK_REPLACEMENT is defined) to CompressDebugInfo:: RestoreBoundariesAndVars().

Reply:
The R2R debuginfo data do not have the OSR flag byte. So when this is called from `ReadyToRunJitManager::EnumMemoryRegionsForMethodDebugInfo`, the check for the flag should be shipped.

The fix should be to add `BOOL hasFlagByte` to this method, similar to how it is done for `CompressDebugInfo::RestoreBoundariesAndVars`

Reply:
I see now. I assumed there was only one call to  CompressDebugInfo::EnumMemoryRegions